### PR TITLE
Move system information gathering to common crate

### PR DESCRIPTION
**Pull request by onalante-msft**
`edgelet` uses the `OsInfo`, `DmiInfo`, and `parse_shell_line` functions.

### CODE REVIEW

File: aziotctl/aziotctl-common/src/host_info.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+    Some((key, value))
+}
+
+fn try_read_dmi(entry: &'static str) -> Option<String> {
+    let path = format!("/sys/devices/virtual/dmi/id/{}", entry);
+
+    let bytes = fs::read(path).ok()?;
+
+    Some(String::from_utf8(bytes)
```
Comment (arsing) [inline aziotctl/aziotctl-common/src/host_info.rs:141]: If you're `to_owned()`'ing at the end anyway, you might as well do `str::from_utf8()` and save on the first copy.

File: aziotctl/Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ foreign-types-shared = "0.1"
 hyper = "0.14"
 hyper-openssl = "0.9"
 libc = "0.2"
-nix = "0.18"
+nix = "0.22"
```
Comment (arsing) [inline aziotctl/Cargo.toml:21]: Why is this needed?

And you might as well update to 0.23 then (the current latest).

**Review by arsing — APPROVED** [review]

File: aziotctl/aziotctl-common/src/host_info.rs [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+pub fn parse_shell_line(line: &str) -> Option<(&str, &str)> {
+    let line = line.trim();
+
+    let (key, value) = line.split_once('=')?;
+
+    // The value is essentially a shell string, so it can be quoted in single or
+    // double quotes, and can have escaped sequences using backslash.
+    // For simplicitly, just trim the quotes instead of implementing a full shell
+    // string grammar.
+    let value = if (value.starts_with('\'') && value.ends_with('\''))
```
Comment (gordonwang0) [inline aziotctl/aziotctl-common/src/host_info.rs:125]: `trim_matches` could replace this block.

**Review by gordonwang0 — APPROVED** [review]

File: aziotctl/aziotctl-common/src/host_info.rs (hunk 3 above)
Comment (arsing) [reply aziotctl/aziotctl-common/src/host_info.rs:125]: I think I wrote it this way to ensure the quotes are matched up on either end.

Comment (gordonwang0) [reply aziotctl/aziotctl-common/src/host_info.rs:125]: Got it. Yeah, that makes sense.
